### PR TITLE
Set annotation name in federated replica set controller

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @quinton-hoole @jianhuiz @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31223)

### CODE REVIEW

File: federation/pkg/federation-controller/replicaset/replicasetcontroller.go [hunk 1]
```diff
@@ -57,15 +56,7 @@ var (
...
-	accessor, err := meta.Accessor(frs)
-	if err != nil {
-		return nil, err
-	}
-	anno := accessor.GetAnnotations()
-	if anno == nil {
-		return nil, nil
-	}
-	frsPrefString, found := anno[FedReplicaSetPreferencesAnnotation]
+	frsPrefString, found := frs.Annotations[FedReplicaSetPreferencesAnnotation]
```
Comment (wojtek-t) [inline federation/pkg/federation-controller/replicaset/replicasetcontroller.go]: Please check that annotations are non-nil.

Comment (mwielgus) [inline federation/pkg/federation-controller/replicaset/replicasetcontroller.go]: Yeah, right. Done.

**Comment by jianhuiz** [comment]
I'm confused. Isn't meta.Accessor the preferred way of accessing object meta data?

**Comment by mwielgus** [comment]
I guess it would be if we didn't know the object type.

**Pull request merged**